### DISCUSSION TITLE
How to get the device width in platform-independent-units?

### DISCUSSION THREAD

**Discussion by Dry-Tech** [dotnet/maui#14686] [Q&A]
Hello,
i need to set the width of an element at runtime. I want it to be 90% the width of the screen.

So how can i get the screen width in platform independent units, in order to use that value with WidthRequest?

Thanks.

**Comment by FM1973** [comment]
Hey there.

I´m using something like this:

`var width = DeviceDisplay.Current.MainDisplayInfo.Width / DeviceDisplay.Current.MainDisplayInfo.Density;`

**Reply by TheXenocide** [reply]
This does not appear to be working as expected for me, at least on Android (not sure about iOS, haven't tested there yet). 

I've posted [this question on StackOverflow](https://stackoverflow.com/questions/78380691/net-maui-how-do-i-predict-calculate-widths-and-heights-relative-to-physical-sc) because I'm getting very unexpected measurements.
